buffer.go

```diff
@@ -648,6 +648,43 @@ func (p *bufferedPage) Release() {
 	bufferUnref(p.repetitionLevels)
 }
 
+type bufferedDictionary struct {
+	Dictionary
+	values  *buffer
+	offsets *buffer
+}
+
+func newBufferedDictionary(dict Dictionary, values, offsets *buffer) Dictionary {
+	p := &bufferedDictionary{
+		Dictionary: dict,
+		values:     values,
+		offsets:    offsets,
+	}
+	bufferRef(values)
+	bufferRef(offsets)
+	return p
+}
+
+func (d *bufferedDictionary) Page() Page {
+	return newBufferedPage(
+		d.Dictionary.Page(),
+		d.values,
+		d.offsets,
+		nil,
```

Comment:
would also appreciate some eyes on whether this page construction is correct (without rep/def levels)

Reply:
Yes, these pointers are only needed for memory management, they can be nil when there are no buffers to track.